src/TpetraLinearSystem.C

```diff
@@ -245,17 +244,62 @@ void TpetraLinearSystem::beginLinearSystemConstruction()
   std::vector<stk::mesh::Entity>::iterator iter = std::unique(owned_nodes.begin(), owned_nodes.end(), CompareEntityEqualById(bulkData, realm_.naluGlobalId_));
   owned_nodes.erase(iter, owned_nodes.end());
 
+  ThrowRequire((int)maxOwnedRowId_ == (int)(owned_nodes.size()*numDof_)); // if this is never tripped, then the above unique is useless. !!!
+
+  // use the Contiguous Map constructor. 
+
+   const Teuchos::RCP<LinSys::Comm> tpetraComm = Teuchos::rcp(new LinSys::Comm(bulkData.parallel()));
+  ownedRowsMap_ = Teuchos::rcp(new LinSys::Map(Teuchos::OrdinalTraits<Tpetra::global_size_t>::invalid(),
+                                               maxOwnedRowId_, // must have *numDof_ 
+                                               1,
+                                               tpetraComm));
   myLIDs_.clear();
-  //KOKKOS: Loop noparallel push_back totalGids_ (std::vector)
+  myLIDs_.reserve(maxOwnedRowId_ +  numSharedNotOwnedNotLocallyOwned*numDof_);
+
+  LocalOrdinal localId = 0;
+  int gstart = 0;
+
+  GlobalOrdinal gomin   = ownedRowsMap_->getMinGlobalIndex();
+
+  if(realm_.tpetGlobalId_ == nullptr) {
+    std::cout << " beginLinearSystemConstruction tpetGlobalID_ is null"
+              <<std::endl<<std::flush;
+  }
+  ThrowRequire(realm_.naluGlobalId_ != nullptr);
+  ThrowRequire(realm_.tpetGlobalId_ != nullptr);
+
   for(stk::mesh::Entity entity : owned_nodes) {
     const stk::mesh::EntityId entityId = *stk::mesh::field_data(*realm_.naluGlobalId_, entity);
     myLIDs_[entityId] = numDof_*localId++;
-    for(unsigned idof=0; idof < numDof_; ++ idof) {
-      const GlobalOrdinal gid = GID_(entityId, numDof_, idof);
-      ownedGids.push_back(gid);
-    }
+    auto *  thisgid = stk::mesh::field_data(*realm_.tpetGlobalId_, entity);
+    ThrowRequire(thisgid != nullptr);
```

Comment:
Is this throw really necessary? Also should this and the next one be `ThrowAssert`? @alanw0  comments?